lib/Client.js

```diff
@@ -188,6 +188,14 @@ class Client extends EventEmitter {
             }
         }
 
+        if(token === undefined) {
```

Comment:
As it is, this check is redundant since `typeof undefined !== "string"` - it would be handled below.

If we're just adding runtime checks for the sake of runtime checks, can't hurt to take it further with `!token` to also check for empty strings. Might also make sense to do both checks at once and have a single error message ("Invalid token ${token}" or something like that would probably work well enough).

Reply:
fair ill do this

Reply:
there are certain circumstances where someone doesn't need to provide a token, so I don't think we should toss out empty strings

Reply:
Is it valid to just not pass a token at all (resulting in `undefined`) in those cases?

Reply:
oop okay

Reply:
> Is it valid to just not pass a token at all (resulting in `undefined`) in those cases?

yes, which is likely why we don't have specific checks, but we can't really do types like that, so checks should in theory be put in place

honestly a specific rest/webhook only client would solve that issue, but that's a bit overkill

Reply:
undefined already errors out aswell so